lib/src/command/lish.dart

```diff
@@ -251,64 +271,120 @@ the \$PUB_HOSTED_URL environment variable.''',
       usageException('Cannot use both --force and --dry-run.');
     }
 
-    if (entrypoint.root.pubspec.isPrivate) {
+    if (_fromArchive != null && _toArchive != null) {
+      usageException('Cannot use both --from-archive and --to-archive.');
+    }
+
+    if (_fromArchive != null && dryRun) {
+      usageException('Cannot use both --from-archive and --dry-run.');
+    }
+
+    if (_toArchive != null && dryRun) {
+      usageException('Cannot use both --to-archive and --dry-run.');
```

Comment:
Why not, just don't create the archive? It's effectively the same as `--dry-run` without `--to-archive`.

Reply:
Yeah why give a file-name if you don't want the file. It indicates you have broken logic somewhere.